### PR TITLE
Added TempData Unit Tests.

### PR DESCRIPTION
**Pull request by IvayloGoranov**

### CODE REVIEW

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 1]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .Ok();
+                },
+                "When calling AddTempDataAction action in MvcController expected temp data to have entry with 'Invalid' key and the provided value, but such was not found.");
+        }
+
+        [Fact]
+        public void TempDataWithBuilderWithPredicateShouldWorkWithCorrectPassingAssertions()
+        {
+            MyController<MvcController>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 2]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .AndAlso()
+                        .ShouldReturn()
+                        .Ok();
+        }
+
+        [Fact]
+        public void TempDataWithBuilderWithPredicateShouldWorkWithCorrectPassingPredicate()
+        {
+            MyController<MvcController>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 3]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .Ok();
+                },
+                "When calling AddTempDataAction action in MvcController expected temp data to have entry with 'Test' key and value passing the given predicate, but it failed.");
+        }
+
+        [Fact]
+        public void ContainingEntryOfTypeShouldNotThrowWithCorrectEntry()
+        {
+            MyController<MvcController>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 4]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .Ok();
+                },
+                "When calling AddTempDataAction action in MvcController expected temp data to have at least one entry of Int32 type, but none was found.");
+        }
+
+        [Fact]
+        public void ContainingEntryOfTypeAndKeyShouldNotThrowWithCorrectEntry()
+        {
+            MyController<MvcController>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 5]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .Ok();
+                },
+                "When calling AddTempDataAction action in MvcController expected temp data to have entry with 'Test' key and value of Int32 type, but in fact found String.");
+        }
+
+        [Fact]
+        public void ContainingEntryWithValueShouldNotThrowWithCorrectEntry()
+        {
+            MyController<MvcController>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 6]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .Ok();
+                },
+                "When calling AddTempDataAction action in MvcController expected temp data to have entry with the provided value, but none was found.");
+        }
+
+        [Fact]
+        public void ContainingEntryWithKeyShouldNotThrowWithCorrectEntry()
+        {
+            MyController<MvcController>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ActionsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ControllersTests/ControllerBuilderTests.cs [hunk 7]
```diff
@@ -22,6 +22,20 @@ public void WithTempDataShouldPopulateTempDataCorrectly()
                 .Ok();
         }
 
+        [Fact]
+        public void WithInvalidTempDataValueShouldReturnBadRequest()
+        {
+            MyController<MvcController>
+                .Instance()
+                .WithTempData(tempData =>
+                {
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ControllersTests/ControllerBuilderTests.cs]: Remove the curly brackets as the code can be written in one line.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ControllersTests/ControllerBuilderTests.cs [hunk 8]
```diff
@@ -58,6 +72,29 @@ public void WithTempDataShouldPopulateTempDataCorrectlyForPocoController()
...
+                .StartsFrom<DefaultStartup>()
+                .WithServices(services =>
+                {
+                    services.TryAddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+                });
+
+            MyController<FullPocoController>
+                .Instance()
+                .WithTempData(tempData =>
+                {
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ControllersTests/ControllerBuilderTests.cs]: Remove the curly brackets here as the code can be written with one line.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 9]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .View();
+                },
+                "When invoking AddTempDataComponent expected temp data to have entry with 'Invalid' key and the provided value, but such was not found.");
+        }
+
+        [Fact]
+        public void TempDataWithBuilderWithPredicateShouldWorkWithCorrectPassingAssertions()
+        {
+            MyViewComponent<AddTempDataComponent>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 10]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .AndAlso()
+                        .ShouldReturn()
+                        .View();
+        }
+
+        [Fact]
+        public void TempDataWithBuilderWithPredicateShouldWorkWithCorrectPassingPredicate()
+        {
+            MyViewComponent<AddTempDataComponent>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 11]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .View();
+                },
+                "When invoking AddTempDataComponent expected temp data to have entry with 'Test' key and value passing the given predicate, but it failed.");
+        }
+
+        [Fact]
+        public void ContainingEntryOfTypeShouldNotThrowWithCorrectEntry()
+        {
+            MyViewComponent<AddTempDataComponent>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 12]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .View();
+                },
+                "When invoking AddTempDataComponent expected temp data to have at least one entry of Int32 type, but none was found.");
+        }
+
+        [Fact]
+        public void ContainingEntryOfTypeAndKeyShouldNotThrowWithCorrectEntry()
+        {
+            MyViewComponent<AddTempDataComponent>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 13]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .View();
+                },
+                "When invoking AddTempDataComponent expected temp data to have entry with 'Test' key and value of Int32 type, but in fact found String.");
+        }
+
+        [Fact]
+        public void ContainingEntryWithValueShouldNotThrowWithCorrectEntry()
+        {
+            MyViewComponent<AddTempDataComponent>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs [hunk 14]
```diff
@@ -131,5 +132,416 @@ public void TempDataWithBuilderShouldWorkCorrectly()
...
+                        .View();
+                },
+                "When invoking AddTempDataComponent expected temp data to have entry with the provided value, but none was found.");
+        }
+
+        [Fact]
+        public void ContainingEntryWithKeyShouldNotThrowWithCorrectEntry()
+        {
+            MyViewComponent<AddTempDataComponent>
+                        .Instance()
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/InvocationsTests/ShouldHaveTests/ShouldHaveTempDataTests.cs]: Wrong indentation here.

File: test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ViewComponentsTests/ViewComponentBuilderTests.cs [hunk 15]
```diff
@@ -22,6 +22,20 @@ public void WithTempDataShouldPopulateTempDataCorrectly()
                 .Content("value");
         }
 
+        [Fact]
+        public void WithInvalidTempDataValueShouldReturnViewWithNoModel()
+        {
+            MyViewComponent<TempDataComponent>
+                .Instance()
+                .WithTempData(tempData =>
+                {
```
Comment (ivaylokenov) [inline test/MyTested.AspNetCore.Mvc.TempData.Test/BuildersTests/ViewComponentsTests/ViewComponentBuilderTests.cs]: Wrong indentation here.

**Comment by IvayloGoranov** [comment]
Indentation fixes applied.

**Pull request merged**